components/application-operator/pkg/kymahelm/application/releasemanager.go

```diff
@@ -48,6 +56,37 @@ func (r *releaseManager) InstallChart(application *v1alpha1.Application) (hapi_4
 	return installResponse.Release.Info.Status.Code, installResponse.Release.Info.Description, nil
 }
 
+func (r *releaseManager) UpgradeReleases() error {
+	appList, err := r.appClient.List(v1.ListOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "Error fetching application list")
+	}
+
+	for _, app := range appList.Items {
+		status, description, err := r.upgradeChart(&app)
+		if err != nil {
+			setCurrentStatus(&app, status.String(), description)
+			r.appClient.Update(&app)
+		}
+	}
+
+	return nil
+}
+
+func (r *releaseManager) upgradeChart(application *v1alpha1.Application) (hapi_4.Status_Code, string, error) {
+	overrides, err := r.prepareOverrides(application)
+	if err != nil {
+		return hapi_4.Status_FAILED, "", errors.Wrapf(err, "Error parsing overrides for %s Application", application.Name)
+	}
+
+	upgradeResponse, err := r.helmClient.UpdateReleaseFromChart(applicationChartDirectory, application.Name, overrides)
```

Comment:
It would be good to log that we started the updated operation for a specific release and also log if the update fails.